src/components/App/App.tsx

```diff
@@ -1,98 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Feed from '../Feed/Feed';
 import Form from '../Form/Form';
 import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
 import './App.css';
 import { CleanArticle, fetchNewsData, getSentiment } from '../../apiCalls'
 
-interface AppState {
-  articles: CleanArticle[],
-  userSentiment: number | string
-}
 
-interface AppProps {}
+const App = (): JSX.Element => {
 
-class App extends React.Component<AppProps, AppState> {
-  constructor(props: AppProps) {
-    super(props)
-    this.state = {
-      articles: [],
-      userSentiment: ''
-    }
-  }
+  const [ articles, setArticles ] = useState<CleanArticle[]>([])
+  const [ userSentiment, setUserSentiment ] = useState<number | null>(null)
+  const [ error, setError ] = useState('')
 
-  componentDidMount = (): void => {
+  useEffect((): void => {
     fetchNewsData()
-      .then((cleanArticles): any => {
-        this.getSentimentScores(cleanArticles)
-          .then(response => {
+      .then((cleanArticles: CleanArticle[]): void => {
+        getSentimentScores(cleanArticles)
+          .then((response: number[]) => {
 
             const scoredArticles = cleanArticles.map((article, i) => {
                article.sentiment = response[i] || 0
                return article;
             })
 
-            this.setState({ articles: scoredArticles })
+            setArticles(scoredArticles)
           })
       })
-  }
+      .catch(error => setError(error.message))
+  }, [])
 
-  getSentimentScores = (cleanArticles: CleanArticle[]): Promise<any> => {
-    return (Promise as any).all(
-      cleanArticles.map((article: CleanArticle, i: number) => {
+  const getSentimentScores = (cleanArticles: CleanArticle[]): Promise<number[]> => {
+    return Promise.all(
+      cleanArticles.map((article: CleanArticle) => {
         return getSentiment(article.abstract)
       })
     )
   }
 
-  changeUserSentiment = (newSentiment: number) => {
-    this.setState({ userSentiment: newSentiment })
-    this.sortBySentiment(newSentiment)
+  const changeUserSentiment = (newSentiment: number) => {
+    setUserSentiment(newSentiment)
+    sortBySentiment(newSentiment)
   }
 
-  sortBySentiment = (newSentiment: number): void => {
-    const { articles } = this.state;
-    const articlesCopy = articles.slice();
-    let sortedArticles: CleanArticle[];
+  const sortBySentiment = (newSentiment: number): void => {
+    let sortedArticles = articles.slice();
```

Comment:
Do we still need to make a copy like this to avoid directly changing state? Maybe we can do without now that we've implemented Hooks. What do y'all think?
```suggestion
  const sortBySentiment = (newSentiment: number): void => {
    let sortedArticles : CleanArticle[] = [];

    if (newSentiment === -1) {
      sortedArticles = articles.sort((articleA, articleB) => {
        return articleB.sentiment - articleA.sentiment;
      })
    } else if (newSentiment === 1) {
      sortedArticles = articles.sort((articleA, articleB) => {
        return articleA.sentiment - articleB.sentiment;
      })
    }

    setArticles(sortedArticles)
  }
```

Reply:
I think that would work except for when the user clicks neutral. Should we add an else statement for it to sort them randomly when sentiment === 0?

Reply:
Yes, we'd have to do that. Good catch. It might be another line or so of code, but it might improve readability by making our three avenues of sorting more explicit.

Reply:
Btw, @nzambonivergara, if you want to incorporate these changes, would you mind trying out the commit suggestion feature? You might want to add the related suggested changes below to a batch before committing.